nginx/templates/nginx.conf

```diff
@@ -26,39 +26,44 @@ http {
     log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
                       '$status $body_bytes_sent "$http_referer" '
                       '"$http_user_agent" "$http_x_forwarded_for" "$http_x_request_id"';
-
-    log_format logstash_json '{'
-                             '"timestamp_msec": "$msec", '
-                             '"@fields": { '
-                             '"remote_addr": "$remote_addr", '
-                             '"real_ip": "$http_x_real_ip", '
-                             '"real_forwarded_for": "$http_x_forwarded_for", '
-                             '"real_forwarded_proto": "$http_x_forwarded_proto", '
-                             '"request_id": "$http_x_request_id", '
-                             '"remote_user": "$remote_user", '
-                             '"body_bytes_sent": $body_bytes_sent, '
-                             '"request_time": $request_time, '
-                             '"request_uri": "$request_uri", '
-                             '"status": $status, '
-                             '"request": "$request", '
-                             '"request_method": "$request_method", '
-                             '"http_referrer": "$http_referer", '
-                             '"http_user_agent": "$http_user_agent", '
-                             '"bytes_sent": $bytes_sent, '
-                             '"gzip_ratio": "$gzip_ratio", '
-                             '"http_host": "$host", '
-                             '"sent_http_location": "$sent_http_location", '
-                             '"server_name": "$server_name", '
-                             '"server_port": "$server_port", '
-                             '"ssl_protocol": "$ssl_protocol", '
-                             '"ssl_cipher": "$ssl_cipher", '
-                             '"ssl_client_s_dn": "$ssl_client_s_dn", '
-                             '"ssl_session_id": "$ssl_session_id", '
-                             '"upstream_addr": "$upstream_addr", '
-                             '"upstream_response_length": "$upstream_response_length", '
-                             '"upstream_response_time": "$upstream_response_time", '
-                             '"upstream_status": "$upstream_status" '
-                             '} }';
+    {% if nginx.log_format %}
+        {% for k,v in nginx.log_format %}
+            log_format {{ k }} {{ v }}
+        {% end %}
+    {% else %}
+        log_format logstash_json '{'
```

Comment:
This would mean that if *any* log_format entry is specified, we lose logstash_json (unless we re-specify it in the pillar). Maybe better to leave logstash_json in-place, unless its specifically overwritten in the pillar?